oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBConverter.java

```diff
@@ -0,0 +1,265 @@
+        } else if (StorageDataComplexObject.class.isAssignableFrom(clazz)) {
+            return TagAndValue.stringTagValue(((StorageDataComplexObject<?>) value).toStorageData());
+        } else if (Layer.class.equals(clazz)) {
+            return TagAndValue.longTagValue((int) value);
+        } else if (JsonObject.class.equals(clazz)) {
+            return TagAndValue.stringTagValue((String) value);
+        } else if (byte[].class.equals(clazz)) {
+            return TagAndValue.stringTagValue((String) value);
+        }
+        throw new IllegalStateException(clazz.getSimpleName() + " is not supported");
+    }
+
+    private static Serializable<BanyandbModel.FieldValue> buildField(Object value, final Class<?> clazz) {
+        if (Integer.class.equals(clazz) || int.class.equals(clazz)) {
+            return TagAndValue.longFieldValue(((Number) value).longValue());
+        } else if (Long.class.equals(clazz) || long.class.equals(clazz)) {
+            return TagAndValue.longFieldValue((Long) value);
+        } else if (String.class.equals(clazz)) {
+            return TagAndValue.stringFieldValue((String) value);
+        } else if (Double.class.equals(clazz) || double.class.equals(clazz)) {
+            return TagAndValue.binaryFieldValue(ByteUtil.double2Bytes((double) value));
+        } else if (StorageDataComplexObject.class.isAssignableFrom(clazz)) {
+            return TagAndValue.stringFieldValue(((StorageDataComplexObject<?>) value).toStorageData());
+        }
+        throw new IllegalStateException(clazz.getSimpleName() + " is not supported");
+    }
+
+    public static class StorageToMeasure implements Convert2Entity {
+        private final MetadataRegistry.Schema schema;
+        private final DataPoint dataPoint;
+
+        public StorageToMeasure(String modelName, DataPoint dataPoint) {
+            this.schema = MetadataRegistry.INSTANCE.findMetadata(modelName);
+            this.dataPoint = dataPoint;
+        }
+
+        @Override
+        public Object get(String fieldName) {
+            MetadataRegistry.ColumnSpec spec = schema.getSpec(fieldName);
+            switch (spec.getColumnType()) {
+                case TAG:
+                    if (double.class.equals(spec.getColumnClass())) {
+                        return ByteUtil.bytes2Double(dataPoint.getTagValue(fieldName));
+                    } else {
+                        return dataPoint.getTagValue(fieldName);
+                    }
+                case FIELD:
+                default:
+                    if (double.class.equals(spec.getColumnClass())) {
+                        return ByteUtil.bytes2Double(dataPoint.getFieldValue(fieldName));
+                    } else {
+                        return dataPoint.getFieldValue(fieldName);
+                    }
+            }
+        }
+
+        @Override
+        public byte[] getBytes(String fieldName) {
+            // TODO: double may be a field?
```

Comment:
`double`? This is a `getBytes` method only. And bytes could not be as tag, right?